src/common/name-syntax.k

```diff
@@ -1,17 +1,22 @@
 module NAME-SYNTAX
 imports LIST-SYNTAX
-//todo: why does this pure syntax definition affect krun?
-//PackageName, AmbName and PackageorTypeName syntax doesn't cause problems
-                     syntax PackageName ::= IdList [klabel('PackageName)]
-                     syntax AmbName ::= Id [klabel('AmbName)]
-                                      | AmbName "." Id [klabel('AmbName)]
-                     syntax PackageOrTypeName ::= Id [klabel('PackageOrTypeName)]
-                                                | PackageOrTypeName "." Id [klabel('PackageOrTypeName)]//strict(1)
+
+syntax PackageName ::= IdList                           [klabel('PackageName)]
+
+syntax AmbName ::= Id                                   [klabel('AmbName)]
+                 | AmbName "." Id                       [klabel('AmbName)]
+
+syntax PackageOrTypeName ::= Id                         [klabel('PackageOrTypeName)]
+                           | PackageOrTypeName "." Id   [strict(1), klabel('PackageOrTypeName)]
+
+syntax ExprName::= Id                                   [klabel('ExprName)]
+                 | AmbName "." Id                       [klabel('ExprName)]
+
+syntax TypeName ::= Id                                  [klabel('TypeName)]
```

Comment:
Yes, the same;

context-free syntax
    Id             -> AmbName {cons("AmbName")}
    AmbName "." Id -> AmbName {cons("AmbName")}

    Id                       -> TypeName {cons("TypeName")}
    PackageOrTypeName "." Id -> TypeName {cons("TypeName")}

    Id             -> ExprName {cons("ExprName")}
    AmbName "." Id -> ExprName {cons("ExprName")}

    Id             -> MethodName  {cons("MethodName")}
    AmbName "." Id -> MethodName  {cons("MethodName")}

    Id                       -> PackageOrTypeName {cons("PackageOrTypeName")}
    PackageOrTypeName "." Id -> PackageOrTypeName {cons("PackageOrTypeName")}